### PR TITLE
Modified triangle and dot in annotators-core 

### PR DESCRIPTION
**Pull request by dsaha21**
# Description

Adding changes as described in the [ISSUE](https://github.com/roboflow/supervision/issues/1285)

The modifications upon the existing code base might result for a more clear visualisation 

# Testing
Also adding the [colab](https://colab.research.google.com/drive/1vHUSwxYtMz_wbp7IH18kFlZBD1UZdeYV#scrollTo=Gk2KhyBkw6h6) file for checking and testing.

Thanks.

### CODE REVIEW

**Comment by dsaha21** [comment]
I have read the CLA Document and I sign the CLA

File: supervision/annotators/core.py [hunk 1]
```diff
@@ -830,9 +830,10 @@ class DotAnnotator(BaseAnnotator):
     def __init__(
         self,
         color: Union[Color, ColorPalette] = ColorPalette.DEFAULT,
-        radius: int = 4,
+        radius: int = 5,
```
Comment (LinasKo) [inline supervision/annotators/core.py:833]: Let's leave the default value at `4`, even if it is hard to see.
People may already be using this in production systems.

File: supervision/annotators/core.py [hunk 2]
```diff
@@ -847,6 +848,7 @@ def __init__(
         self.radius: int = radius
         self.position: Position = position
         self.color_lookup: ColorLookup = color_lookup
+        self.outer_thickness = outer_thickness
```
Comment (LinasKo) [inline supervision/annotators/core.py:851]: Let's keep the naming the same as much as possible. `outline_thickness` sounds correct here too.

File: supervision/annotators/core.py [hunk 3]
```diff
@@ -1669,27 +1674,80 @@ def annotate(
...
-            tip_x, tip_y = int(xy[detection_idx, 0]), int(xy[detection_idx, 1])
-            vertices = np.array(
-                [
-                    [tip_x - self.base // 2, tip_y - self.height],
-                    [tip_x + self.base // 2, tip_y - self.height],
-                    [tip_x, tip_y],
-                ],
-                np.int32,
-            )
+        if self.outline_thickness:
```
Comment (LinasKo) [inline supervision/annotators/core.py:1677]: This is a lot of logic for a small change.

A more efficient method involve using `cv2.polylines`.
There you can specify thickness and draw another shape behind the original triangle.

I wouldn't expect to have two very different if-else branches.

File: supervision/annotators/core.py [hunk 4]
```diff
@@ -899,6 +901,7 @@ def annotate(
             )
             center = (int(xy[detection_idx, 0]), int(xy[detection_idx, 1]))
             cv2.circle(scene, center, self.radius, color.as_bgr(), -1)
+            cv2.circle(scene, center, self.radius, (0, 0, 0), self.outer_thickness)
```
Comment (LinasKo) [inline supervision/annotators/core.py:904]: I'd expect the background to be drawn first. I'd swap these 2 `cv2.circle` lines.

**Review by LinasKo — CHANGES_REQUESTED** [review]
Good starting point - I like the testing colab and the resulting annotation, but there's way to improve the code. I left a few comments 😉 

I'll review again after that's done.

File: supervision/annotators/core.py (hunk 1 above)
Comment (dsaha21) [reply supervision/annotators/core.py:833]: Cool, will keep it same.

File: supervision/annotators/core.py (hunk 3 above)
Comment (dsaha21) [reply supervision/annotators/core.py:1677]: Actually the original triangles are little small :)
No problem, I will add cv2.polygons and the thickness parameter.

**Comment by onuralpszr** [comment]
@LinasKo I added doc part quickly, and rest looks better but @dsaha21  can you squash your commits and make meaningful comment message, or I can do it too (I am gonna squash couple of them and write something cleaner)

File: supervision/annotators/core.py [hunk 5]
```diff
@@ -898,6 +901,7 @@ def annotate(
                 else custom_color_lookup,
             )
             center = (int(xy[detection_idx, 0]), int(xy[detection_idx, 1]))
+            cv2.circle(scene, center, self.radius, (0, 0, 0), self.outline_thickness)
```
Comment (LinasKo) [inline supervision/annotators/core.py:904]: Let's only do this when `self.outline_thickness` is not 0, just like in the triangle annotator

**Review by LinasKo — APPROVED** [review]

File: supervision/annotators/core.py (hunk 5 above)
Comment (dsaha21) [reply supervision/annotators/core.py:904]: Yes @LinasKo, will add a condition statement here. Updating it and submiting.

**Comment by dsaha21** [comment]
Hi @LinasKo, please check the latest [colab notebook](https://colab.research.google.com/drive/1vHUSwxYtMz_wbp7IH18kFlZBD1UZdeYV#scrollTo=Gk2KhyBkw6h6)

These are the final changes. Please let me know about further changes if needed. Thanks

**Comment by LinasKo** [comment]
Thank you, @dsaha21 and @onuralpszr!

It's great to have such a nice contribution in the repo this quick 😉

**Pull request merged**